gatling-http/src/main/scala/io/gatling/http/util/package.scala

```diff
@@ -107,5 +110,18 @@ package object util extends LazyLogging {
           }
         case _ => buff
       }
+
+    def appendWebsocketInboundMessages(messageBuffer: Seq[(Long, String)]): jl.StringBuilder = {
+      buff.append(
+        messageBuffer.zipWithIndex
+          .map { case ((timestamp, message), index) =>
+            val time = bufferFormat.format(timestamp)
+            s"$time [$index] -> $message"
+          }
+          .mkString("\n")
```

Comment:
If you use a loop to repeatedly `append` to the `StringBuilder`, you can save the allocation of

- the `List` with indices,
- the `List` of `String`s
- the final concatenated `String`.

Also it looks like you need the final new line anyway.

Reply:
@phiSgr What do I need to use instead of `map`?

Reply:
Use [`foreach`](https://www.scala-lang.org/api/2.13.x/scala/collection/Seq.html#foreach[U](f:A=%3EU):Unit) for effects.

This principle also applies in many other languages like [JS](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/forEach) and [Java](https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#forEach-java.util.function.Consumer-).

Reply:
now I see) thx)
Oh, I know about `foreach`, but I don't know why I forgot it 🙈